src/contract/deploy/impl/DefaultCreateContract.ts

```diff
@@ -46,23 +57,24 @@ export class DefaultCreateContract implements CreateContract {
     srcTx: Transaction = null
   ): Promise<ContractDeploy> {
     this.logger.debug('Creating new contract from src tx');
-    const { wallet, srcTxId, initState, tags, transfer, data } = contractData;
+    const { srcTxId, initState, tags, transfer, data } = contractData;
+    if (!this.wallet.signature) {
+      this.wallet.getSignature(contractData.wallet);
+    }
+    const signer = this.wallet.signature.signer;
 
     const effectiveUseBundler =
       disableBundling == undefined ? this.warp.definitionLoader.type() == 'warp' : !disableBundling;
 
-    let contractTX = await this.arweave.createTransaction({ data: data?.body || initState }, wallet);
```

Comment:
not sure about this one - can we skip the wallet here reference here and just leave it when signing? seems to work.

Reply:
I believe all it does is https://github.com/ArweaveTeam/arweave-js/blob/master/src/common/common.ts#L98

But this is also done by the `sign` (https://github.com/ArweaveTeam/arweave-js/blob/master/src/common/transactions.ts#L231) method, so should be fine.